go/arrow/compute/internal/exec/span.go

```diff
@@ -85,16 +86,20 @@ type ArraySpan struct {
 	Children []ArraySpan
 }
 
+func (a *ArraySpan) MayHaveNulls() bool {
+	return atomic.LoadInt64(&a.Nulls) != 0 && a.Buffers[0].Buf != nil
+}
+
 // UpdateNullCount will count the bits in the null bitmap and update the
 // number of nulls if the current null count is unknown, otherwise it just
 // returns the value of a.Nulls
 func (a *ArraySpan) UpdateNullCount() int64 {
 	if a.Nulls != array.UnknownNullCount {
```

Comment:
Should this loading of  `a.Nulls` also be atomic to prevent data race?

Reply:
Good catch, i'll fix that.